modular_skyrat/modules/supermatter_glow_component/glow.dm

```diff
@@ -0,0 +1,95 @@
+#define SUPERMATTER_CASCADE_COLOUR "#333333"
+#define SUPERMATTER_TESLA_COLOUR "#00ffff"
+#define SUPERMATTER_SINGULARITY_RAYS_COLOUR "#750000"
+#define SUPERMATTER_SINGULARITY_LIGHT_COLOUR "#400060"
+
+
+/datum/component/supermatter_glow
+	var/light_color = SUPERMATTER_COLOUR // Nice and yellow.
+	var/obj/machinery/power/supermatter_crystal/our_supermatter // Typecasted reference to our supermatter(parent)
+
+/datum/component/supermatter_glow/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_SM_PROCESS_ATMOS, .proc/update_effects) // When we are added, we listen for this signal and send the proc.
+	our_supermatter = parent // typecast our parent
+
+/datum/component/supermatter_glow/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_SM_PROCESS_ATMOS)
+	our_supermatter = null // no hard del 4 u
+
+/// This proc builds a list of effects to apply to the supermatter based on it's contexts.
+/datum/component/supermatter_glow/proc/update_effects()
+	SIGNAL_HANDLER
+	var/list/filters_to_add = list() // List to apply later.
+/*
+You might be wondering what this does. We're building a list. Filters are an array, right. The first will over-write the first when reapplied..
+However, they do not remove animate() from the entry in the array UNLESS it was deleted entirely.
+So, our first filter is the rays. The second filter will APPLY to the rays, and so forth.
+Sort of like photoshop. The loop will continue forever unless an accident happens.
+*/
+	if(our_supermatter.internal_energy || our_supermatter.damage && !our_supermatter.final_countdown)
+		our_supermatter.set_light((initial(our_supermatter.light_range) + our_supermatter.internal_energy/200), initial(our_supermatter.light_power) + our_supermatter.internal_energy/1000, (our_supermatter.gas_heat_power_generation > 0.8 ? SUPERMATTER_RED : SUPERMATTER_COLOUR), TRUE)
+		filters_to_add |= filter(type="rays", size = clamp(our_supermatter.internal_energy/30, 1, 125), color = (our_supermatter.gas_heat_power_generation > 0.8 ? SUPERMATTER_RED : SUPERMATTER_COLOUR), factor = clamp(our_supermatter.damage/600, 1, 10), density = clamp(our_supermatter.damage/10, 12, 100))
+
+	switch(check_special_delamination())
+		if(SINGULARITY_DELAMINATION)
+			var/rays_filter = filter(type="rays", size = clamp((our_supermatter.damage/100)*our_supermatter.internal_energy, 50, 125), color = SUPERMATTER_SINGULARITY_RAYS_COLOUR, factor = clamp(our_supermatter.damage/300, 1, 30), density = clamp(our_supermatter.damage/5, 12, 200))
+			var/outline_filter = filter(type = "outline", size = 1, color = SUPERMATTER_SINGULARITY_LIGHT_COLOUR)
+
+			if(our_supermatter.final_countdown)
+				var/icon/singulo = new/icon('icons/effects/96x96.dmi', "singularity_s3", frame = rand(1,8)) // Summon the icon as an object
+				var/singulo_filter = filter(type="layer", icon = singulo, flags = FILTER_OVERLAY) // apply the icon object as a layer filter
+				filters_to_add |= list(singulo_filter) // send it to the list
```

Comment:
Would this not be better as a static icon definition, so that we only really create this filter once, as I don't think we need to create it more than once (to avoid adding that filter multiple times to `filters_to_add`?
Also add a space between your `new` and the typepath of the thing you're calling `new` on, for readability purposes.

Reply:
Can't due to filters not animating icons for some reason